engine/src/image.cpp

```diff
@@ -951,6 +953,17 @@ Exec_stat MCImage::getprop_legacy(uint4 parid, Properties which, MCExecPoint& ep
             else
                 ep . clear();
         break;
+        // MERG-2014-09-18: [[ ImageMetadata ]] return the image metadata as an array
+    case P_METADATA:
+        if (m_rep != nil)
+        {
+            MCImageMetadata t_metadata;
+            (static_cast<MCCachedImageRep *>(m_rep))->GetMetadata(t_metadata);
+            MCImageGetMetadata(ep, t_metadata);
```

Comment:
This doesn't cover the case of MCMutableImageRep, which is not a cached image rep. Why not just make GetMetaData a pure virtual method of MCImageRep avoiding the need to cast altogether? The MCMutableImageRep implementation can return false, which you would need to check for and handle here.